app/src/main/java/com/gst/gusto/review/adapter/InstaReviewAdapter.kt

```diff
@@ -0,0 +1,98 @@
+package com.gst.gusto.review.adapter
+
+
+import android.content.Context
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.gst.gusto.R
+import com.gst.gusto.databinding.ItemBufferingBinding
+import com.gst.gusto.databinding.ItemReviewGalleryBinding
+import com.gst.gusto.model.InstaReview
+import com.gst.gusto.review.adapter.viewholder.InstaViewHolder
+import com.gst.gusto.review.adapter.viewholder.LoadingViewHolder
+
+class InstaReviewAdapter(
+    var items: MutableList<InstaReview?>?,
+    private val context: Context?,
+    private val itemClickListener: (Int) -> Unit
+) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
+
+    override fun onCreateViewHolder(viewGroup: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
+        return when(viewType) {
+            R.layout.item_buffering -> {
+                LoadingViewHolder(ItemBufferingBinding.inflate(LayoutInflater.from(viewGroup.context),
+                    viewGroup, false)
+                )
+            }
+            R.layout.item_review_gallery -> {
+                InstaViewHolder(ItemReviewGalleryBinding.inflate(LayoutInflater.from(viewGroup.context),
+                    viewGroup, false),
+                    itemClickListener
+                )
+            }
```

Comment:
여기서 position -> reviewId 변환 작업이 가능해요.
어댑터에서도 InstaReview 에 대한 리스트를 가지고 있으니까요.
```suggestion
            R.layout.item_review_gallery -> {
                val binding = ItemReviewGalleryBinding.inflate(
                    LayoutInflater.from(viewGroup.context, viewGroup, false)
                )
                InstaViewHolder(binding) { position ->
                    itemClickListener(items[position].reviewId) // 이르케
                }
            }
```